.github/workflows/checks.yml

```diff
@@ -16,6 +19,6 @@ jobs:
           scarb-version: 2.3.0
```

Comment:
Ideally, we'd use some kind of "latest compatible" version of scarb instead of fixed, but it's not easily doable: Starknet Foundry is often not immediately compatible with a "latest" scarb release.

Reply:
would it be possible to run setup-snfoundry first, check on its scarb version, and then use this output to setup scarb? or does snfoundry action prohibit this?

Reply:
I think this could be done. I'll try this

Reply:
Starknet Foundry doesn't output a supported scarb version so resolving this is not trivial. I'd keep it fixed for now.